### DISCUSSION TITLE
Hello, how to achieve multi-task management through block code?

### DISCUSSION THREAD

**Discussion by myhewen** [pybricks/support#1578] [Q&A]
For example, Task 1 is to send 8 signals to control 8 trains; Task 2 is to transmit 12 signals to control 12 bifurcated rails; To switch to different tasks through the left and right buttons on the prime HUB or press the buttons on the xbox controller to switch tasks, could you please give an example of block code Settings? Thank you!

At present, the maximum byte limit of the transmitted signal is 26. Can this limit be exceeded by switching tasks?

**Comment by laurensvalk** [comment]
It's best to pack your data into one list or byte array. Sending 8 separate speed values (-100 to 100) costs 16 bytes. If some trains run at the same speed, maybe you don't need 8 values. And then you can send 10 binary values for the switches. But if you want more, you could encode up to 8 switch position in a single numeric value, so in theory up to 40 switches :smile: 

This is quite (!) a bit beyond the hardware that I have --- would it make sense to start testing with a smaller setup and expand from there?

By the way, did the programs in https://github.com/orgs/pybricks/discussions/1546 work for you?

**Comment by myhewen** [comment]
Thank you! But I still want to learn how to multitask Settings and multitask switching, can you give the block code Settings picture?
